src/client-side-metrics/metric-interceptor.ts

```diff
@@ -49,16 +49,21 @@ function createMetricsInterceptorProvider(
             collector.onStatusMetadataReceived(
               status as unknown as ServerStatus,
             );
+            collector.onAttemptComplete(status.code);
             nextStat(status);
           },
         };
         next(metadata, newListener);
       },
+      sendMessage: function (message, next) {
+        collector.onAttemptStart();
+        next(message);
```

Comment:
this is unrelated, but maybe we should actually rename this function to create createMetricsUnaryInterceptorProvider?

I'm realizing we don't seem to mention unary anywhere in this file, and that's actually pretty important, because iirc sendMessage wouldn't be a good place to start the attempt for stream calls